### PR TITLE
Add travis configuration

### PR DESCRIPTION
**Pull request by appinteractive**
Todos:
- [x] change `humanconnection/webapp` to `humanconnection/nitro-web`
- [x] push first image manually

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
+jobs:
+  include:
+    - stage: Test and Build
+      script:
+      - cd styleguide && yarn install --production=false --frozen-lockfile --non-interactive --ignore-engines
+      - yarn run styleguide:build
+      - yarn install --frozen-lockfile --non-interactive --ignore-engines
+      # - yarn run ci
+    - script:
+      - docker build --build-arg BUILD_COMMIT=$TRAVIS_COMMIT -t humanconnection/webapp .
```
Comment (roschaefer) [inline .travis.yml]: ` -t humanconnection/webapp` are you sure? Shouldn't it be rather ` -t humanconnection/nitro-webapp`? Shall we start uploading images to `webapp` after we drop the legacy code?

Comment (appinteractive) [reply .travis.yml]: Would be fine for me, but before we need to push it manually

**Comment by roschaefer** [comment]
Can I suggest a differen approach? We want to build the docker image on Travis anyways. So how about running tests *from within the docker container*? If we deploy the image to production, the docker container should be authoritative for the success of the build.

**Comment by appinteractive** [comment]
Yeah sure, but what about image size?

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,48 @@
...
+  - wget https://raw.githubusercontent.com/DiscordHooks/travis-ci-discord-webhook/master/send.sh
+  - chmod +x send.sh
+  - ./send.sh success $WEBHOOK_URL
+  - if [ $TRAVIS_BRANCH == "master" ] && [ $TRAVIS_EVENT_TYPE == "push" ]; then
+    docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD" &&
+    docker tag humanconnection/webapp humanconnection/webapp:latest &&
+    docker push humanconnection/webapp:latest &&
+    wget https://raw.githubusercontent.com/Human-Connection/Discord-Bot/develop/tester.sh &&
+    chmod +x tester.sh &&
+    ./tester.sh nitro-alpha $WEBHOOK_URL;
```
Comment (roschaefer) [inline .travis.yml]: See https://github.com/Human-Connection/Discord-Bot/issues/16

File: .travis.yml (hunk 1 above)
Comment (roschaefer) [reply .travis.yml]: `webapp` is there but not used by any of our services. I'm fine to use `webapp`.

**Comment by roschaefer** [comment]
Does a command run in the docker container affect the size of the image?? :astonished:

File: .travis.yml (hunk 2 above)
Comment (appinteractive) [reply .travis.yml]: But that does not block us right?

**Comment by appinteractive** [comment]
Not the command but don’t we have then reports and so on?

File: Dockerfile [hunk 3]
```diff
@@ -1,22 +1,27 @@
 FROM node:10-alpine
-LABEL Description="This image builds and runs the Human-Connection Web Application Frontend" Vendor="Human-Connection gGmbH" Version="0.0.1" Maintainer="Human-Connection gGmbH (developer@human-connection.org)"
+LABEL Description="Web Frontend of the Social Network Human-Connection.org" Vendor="Human-Connection gGmbH" Version="0.0.1" Maintainer="Human-Connection gGmbH (developer@human-connection.org)"
 
-# expose the app port
+# Expose the app port
 EXPOSE 3000
 
-ARG WORKDIR=/HC-WebApp
```
Comment (roschaefer) [inline Dockerfile]: I would not add `nitro` here. That name will be removed once we drop the legacy web app.

File: .travis.yml [hunk 4]
```diff
@@ -1,13 +1,30 @@
...
+  - docker
 cache:
+  yarn: true
   directories:
-    - "node_modules"
+    - node_modules
 
-install:
-  - yarn install
+before_script:
```
Comment (roschaefer) [inline .travis.yml]:
```suggestion
install:
```
@appinteractive I guess if you remove `install` entirely, Travis will run `yarn install` by default. What happens if we simply build the docker image during `install`?

**Comment by appinteractive** [comment]
I fixed the linting... will see if its working now.

**Comment by roschaefer** [comment]
One thing: Does `docker run [secure]/nitro-web yarn run lint` exit with non-zero code (thus failing the build), if `yarn run lint` exits with non-zero code in the container? Did you try?

**Pull request merged**